packages/flutter/lib/src/material/expansion_panel.dart

```diff
@@ -121,8 +170,94 @@ class ExpansionPanelList extends StatelessWidget {
   /// The duration of the expansion animation.
   final Duration animationDuration;
 
+  //Whether multiple panels can be open simultaneously
+  final bool _allowMultiplePanelsOpen;
+
+  @override
+  State<StatefulWidget> createState() => new _ExpansionPanelListState();
+}
+
+class _ExpansionPanelListState extends State<ExpansionPanelList> {
+  Map<int, bool> _widgetOpenMap;
+
+  @override
+  void initState() {
+    super.initState();
+
+    if (!widget._allowMultiplePanelsOpen) {
+      _widgetOpenMap = <int, bool>{};
+
+      for (int i = 0; i < widget.children.length; i += 1) {
+        assert(widget.children[i] is ExpansionPanelRadio,
+               'All children of ExpansionPanel.radio need to be of type ExpansionPanelRadio');
+        final ExpansionPanelRadio _widgetChild = widget.children[i];
+        _widgetOpenMap[_widgetChild.value] = _widgetChild.initializesExpanded;
+      }
+
+      assert(_debugCheckExpandLimit(_widgetOpenMap),
+             'This expansion panel widget initialized with more than one panel open');
+    }
+  }
+
+  @override
+  void didUpdateWidget(ExpansionPanelList oldWidget) {
+    super.didUpdateWidget(oldWidget);
+
+    if (!widget._allowMultiplePanelsOpen) {
+      final Map<int, bool> newMap = <int, bool>{};
+      for (int i = 0; i < widget.children.length; i += 1) {
+        final ExpansionPanelRadio child = widget.children[i];
+        final int childKey = child.value;
+        newMap[childKey] = _widgetOpenMap[childKey] ?? child.initializesExpanded;
+
+        assert(widget.children[i] is ExpansionPanelRadio,
+        'All children of ExpansionPanel.radio need to be of type ExpansionPanelRadio');
+      }
+      _widgetOpenMap = newMap;
+      assert(_debugCheckExpandLimit(_widgetOpenMap),
+             'This expansion panel widget initialized with more than one panel open');
+    }
+  }
+
+  //Used for initializing the radio type list.
+  //Checks whether more than one panel is set to expanded.
```

Comment:
Describe what this check does and when it applied:

Checks whether more than one panel is expanded when a radio expansion panel is used, or something like that